### PR TITLE
Fix #5: Use different names for draft implementations.

### LINKED ISSUE #5: Specify that implementations of drafts should include draft number in headers and content encodings.
For example, an implementation of [draft 02](https://tools.ietf.org/html/draft-thomson-http-mice-02) might use the `MI-Draft2` header and the `mi-sha256-draft2` content encoding.

### PR DESCRIPTION
**Pull request by jyasskin**
I'm not tied at all to the particular names here, or where they're expressed within the draft.

### CODE REVIEW

File: draft-thomson-http-mice.md [hunk 1]
```diff
@@ -159,6 +159,14 @@ A message that has a content length less than or equal to the content size does
 not include any inline proofs.  The proof for a message with a single record is
 simply the hash of the body plus a trailing zero octet.
 
+*RFC EDITOR: Please remove the next paragraph before publication.*
+
+Implementations of drafts of this specification MUST implement a content
+encoding named "mi-sha256-draft#" instead of the "mi-sha256" content encoding
```
Comment (martinthomson) [inline draft-thomson-http-mice.md]: mi-sha256-03 for draft 3

Comment (jyasskin) [reply draft-thomson-http-mice.md]: Done.

File: draft-thomson-http-mice.md [hunk 2]
```diff
@@ -273,6 +280,11 @@ which they were applied.
 The MI header MAY be omitted if the sender intends for the receiver to acquire
 the integrity proof for the first record by other means.
 
+*RFC EDITOR: Please remove the next paragraph before publication.*
+
+Implementations of drafts of this specification use the "MI" header field
```
Comment (jyasskin) [inline draft-thomson-http-mice.md]: I left this so @nyaxt et. al. don't have to wonder if we just forgot this part, but I can also just tell them. Let me know if I should undo it.

Comment (martinthomson) [reply draft-thomson-http-mice.md]: It's going anyway.

**Pull request merged**